vrx_gz/src/GymkhanaScoringPlugin.cc

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gz/msgs/param.pb.h>
+#include <gz/msgs/vector3d.pb.h>
+#include <limits>
+#include <gz/sim/Entity.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/Util.hh>
+#include <gz/sim/World.hh>
+#include <gz/plugin/Register.hh>
+#include <gz/transport/Node.hh>
+#include <sdf/sdf.hh>
+
+#include "GymkhanaScoringPlugin.hh"
+
+using namespace gz;
+using namespace vrx;
+
+/// \brief Private GymkhanaScoringPlugin data class.
+class GymkhanaScoringPlugin::Implementation
+{
+  /// \brief Callback for black box station-keeping portion's scoring plugin
+  public: void BlackboxCallback(const msgs::Param &_msg);
```

Comment:
Could you document the `_msg` parameter?

Reply:
Done! I realized our taskMsg is not documented anywhere so I also added this to `ScoringPlugin.cc`

Reply:
Great! Thanks